src/snyk/snykOss/services/vulnerabilityCount/vulnerabilityCountProviderLS.ts

```diff
@@ -158,4 +158,73 @@ export class ModuleVulnerabilityCountProvider {
       return 0;
     })?.[0];
   }
+
+  public getResultArray = (): ReadonlyArray<OssFileResult> | undefined => {
+    if (!this.ossService.result) {
+      return undefined;
+    }
+
+    const tempResultArray: OssFileResult[] = [];
+    const resultCache = new Map<string, OssResultBody>();
+
+    for (const [, value] of this.ossService.result) {
+      // value is Error
+      if (value instanceof Error) {
+        tempResultArray.push(new CliError(value));
+      }
+      // value is Issue<T>[]
+      else {
+        for (const issue of value) {
+          // try to access list of vulns for the current file
+          let res = resultCache.get(issue.filePath);
+
+          // add list of vulns to local cache if not there yet
+          if (res === undefined) {
+            res = {
+              path: issue.filePath,
+              vulnerabilities: [],
+              projectName: issue.additionalData.projectName,
+              displayTargetFile: issue.additionalData.displayTargetFile,
+              packageManager: issue.additionalData.packageManager,
+            };
+            resultCache.set(issue.filePath, res);
+          }
+
+          const tempVuln: OssVulnerability = {
+            id: issue.id,
+            license: issue.additionalData.license,
+            identifiers: issue.additionalData.identifiers,
+            title: issue.title,
+            description: issue.additionalData.description,
+            language: issue.additionalData.language,
+            packageManager: issue.additionalData.packageManager,
+            packageName: issue.additionalData.packageName,
+            severity: issue.severity as unknown as OssSeverity,
```

Comment:
issue: I wouldn't use `as unknown as x` in the main codebase; only to debug/when developing. 

We're overriding the type-system here so `issue.severity` can be anything if we use `as unknown as OssSeverity`.

Probably better to map `IssueSeverity` to `OssSeverity`:

`severity: this.toOssSeverity(issue.severity)`

```
 toOssSeverity(severity: IssueSeverity): OssSeverity {
    switch (severity) {
      case IssueSeverity.Low:
        return OssSeverity.Low;
      case IssueSeverity.Medium:
        return OssSeverity.Medium;
      case IssueSeverity.High:
        return OssSeverity.High;
      default:
        return OssSeverity.Critical;
    }
  }
```

Reply:
Done!